### DISCUSSION TITLE
Show which value exactly is unprocessable

### DISCUSSION THREAD

**Discussion by tweakimp** [fastapi/fastapi#2925] [Questions] [answered]
I have the following code:
```python
import datetime
from typing import List

import uvicorn
from fastapi import FastAPI
from pydantic import BaseModel

app = FastAPI()


class Request(BaseModel):
    times: List[datetime.time]


@app.post("/test")
def test(request: Request):
    return "PASSED"


if __name__ == "__main__":
    uvicorn.run("api:app", host="0.0.0.0", port=8001, reload=True)
```

Now I send the following request:
```
{
  "times": [
    "10:00:00",
    "199:00:00"
  ]
}
```

The response looks like this:
```
{
  "detail": [
    {
      "loc": [
        "body",
        "times",
        1
      ],
      "msg": "invalid time format",
      "type": "value_error.time"
    }
  ]
}
```

The reponse does not tell me which of the values in `times` is unprocessable.
I would be easier to understand the problem if it said `value '199:00:00' has an invalid time format`

**Comment by HarrySky** [comment, marked as answer]
It says in "loc" that problem in "body" -> "times" -> at index 1 :confused:

**Reply by tweakimp** [reply]
Ok, but that is only the index, not the value.

**Reply by HarrySky** [reply]
To get value you check what was sent in "times" at index 1 and see "199:00:00". It returns you all the information you need, there is no need to make it more verbose (increasing bandwidth consumption).

**Reply by kbakk** [reply]
If the value was a credit card number or other personally identifiable information (PII), logging the error with the value could have been a security issue. How it is now, it can be logged without that concern.

**Reply by tweakimp** [reply]
Ok, that is a valid point

**Reply by YuriiMotov** [reply]
And with Pydantic V2 you will find exact value in `input` field

**Comment by ec2ainun** [comment]
@tweakimp 

the msg is clear "invalid time format", check [this out ][1]

```md
    str, following formats work:
        HH:MM[:SS[.ffffff]][Z or [±]HH[:]MM]]]
```

[1]: https://pydantic-docs.helpmanual.io/usage/types/#datetime-types

**Reply by tweakimp** [reply]
I know why the value can not be processed. `99:00` would also fit the format you posted but still be wrong. My suggestion is to specify the value that can not be processed in the returning json to make it easier for the human reader to know which value exactly is invalid.

**Reply by ec2ainun** [reply]
@tweakimp 

it's already easy enough to trace
are you sure `99:00` is valid hour?

**Reply by tweakimp** [reply]
It is not, but in the format you posted it is.

**Reply by ec2ainun** [reply]
@tweakimp 

well, thats format is widely use, and the `H` means hours according to [RFC 3339][1] and [ISO 8601][2]
perhaps you want to change the standard?

[1]: https://tools.ietf.org/html/rfc3339
[2]: https://en.wikipedia.org/wiki/ISO_8601
